src/components/Icon.tsx

```diff
@@ -0,0 +1,27 @@
+import { FunctionComponent, SyntheticEvent } from 'react'
+
+import * as icons from '@mui/icons-material'
+
+interface Icon {
+    name: keyof typeof icons
+    size: number
+    className?: string | undefined
+    color?: string
+    key?: string
+    onClick?: (event: SyntheticEvent) => void
+    variant?: 'boxed'
+}
+
+export const Icon: FunctionComponent<Icon> = ({ name, size, className = '', color, key, onClick, variant }) => {
+    const IconName = icons[name]
+
+    return (
+        <IconName
+            className={`${className} ${variant === 'boxed' ? 'bg-violet-mist text-vivid-violet rounded p-2' : ''}`}
```

Comment:
This is small, but I don't think a ternary is necessary if one result is empty. I'd prefer
```suggestion
className={classNames(className, variant === 'boxed' && 'bg-violet-mist text-vivid-violet rounded p-2')}`
```

or 
```suggestion
className={classNames(className, { 'bg-violet-mist text-vivid-violet rounded p-2': variant === 'boxed' })}
```

Reply:
Thank you for noting this; it is definitely the better way! In addition to your suggestion, I added an import for `classNames`: 31a0f7d.